tests/test_git.py

```diff
@@ -234,7 +234,7 @@ def test_filter():
 
 def test_filter_fails_on_empty_commit_messages():
     with pytest.raises(Exception):
-        _filter_test('', [])
+        _filter_test('', '', [])
```

Comment:
@mrusu91 This was a good catch. Now, the question is why wouldn't this fail when running the unit-tests? And the answer is: "because we are using `pytest.raises(Exception)`, which is too general, and was catching the exception raised by `_filter_test()` being called with the wrong arguments.

I think we should instead replace it with:
```python
 def test_filter_fails_on_empty_commit_messages():
      with pytest.raises(subprocess.CalledProcessError):
          _filter_test('', '', [])
```
and similarly with the other test cases.

Reply:
I fixed this in 1438867, where in fact I check that the error message we get is what we were expecting (or else, the script could be crashing instead). Interestingly, there was a bug in the error handling of `trailfilter` that only surfaced when I tried to do this...